src/battle.test.c

```diff
@@ -437,14 +446,18 @@ static void test_magic_resistance(CuTest *tc)
 
     i_change(&du->items, ishield, 1);
     dt.fighter = setup_fighter(&b, du);
-    calculate_armor(dt, 0, 0, &magres);
+    magres = calculate_resistance(dt,
+        select_weapon(dt, false, true) ? select_weapon(dt, false, true)->type : 0,
```

Comment:
Hier habe ich ein schlechtes Beispiel geliefert: Eigentlich erlaubt C, dass man statt NULL eine 0 schreibt (und manchmal erspart einem das, einen Header einzubinden, in dem NULL definiert ist), aber es ist schlechter Stil, und es gibt zumindest eine Situation, wo man diese Ersetzung nicht machen kann, und über die Eressea schon einmal gestolpert ist (variable argument lists auf 64-bit CPUs). Der doppelte Aufruf von select_weapon in diesem eigentlich unnötigen ternary operator ist auch hässlich, mach das weg :-)